gammapy/cube/fit.py

```diff
@@ -346,6 +346,49 @@ def create(
             gti=gti,
         )
 
+    def stack(self, other):
+        """
+        Stack the maps in other onto self.
+        The stacking is done inplace
+
+        Parameters
+        ----------
+        other: `~gammapy.cube.MapDataset`
+            the MapDatasets to be stacked with this one.
+
+        Returns
+        -------
+        new : `~gammapy.cube.MapDataset`
+            A dataset with the stacked maps
+
+        """
+
+        if self.counts:
+            if other.mask_fit:
+                other.counts = other.counts * other.mask_fit
+            self.counts.coadd(other.counts)
+
+        self.exposure.coadd(other.exposure)
+
+        if other.mask_fit:
+            other.background_model.map = other.background_model.map * other.mask_fit
+        self.background_model.map.coadd(other.background_model.evaluate())
+
+        if isinstance(self.psf, PSFMap):
+            self.psf.stack(other.stack())
+        elif isinstance(other.psf, PSFKernel):
+            raise ValueError("Stacking of PSF kernels not supported")
+
+        if isinstance(other.edisp, EDispMap):
+            self.edisp.stack(other.stack())
+        elif isinstance(self.psf, EnergyDispersion):
+            raise ValueError("Stacking of edisp kernels not supported")
+
+        if self.gti:
+            self.gti = self.gti.stack(other.gti).union()
+
+        # How to stack the mask?
```

Comment:
We should typically want to do logical `or` on the mask as in `SpectrumDataset.stack()`.
However, `mask_fit` is not a `Map` object but a numpy array. How do we take care of the fact that `other.mask_fit` is a cutout of `self.mask_fit` and will not have the same dimensions?

Reply:
This is a good point and probably comes in as an argument to make the `mask_fit` and `mask_safe` a `Map` object as well. For now I think it's OK to temporarily create a map object and use it to stack the mask like so:
```python3

mask_safe = Map.from_geom(self.counts.geom, data=self.mask_safe)
mask_safe_other = Map.from_geom(other.counts.geom, data=other.mask_safe)
```
Note that the `+` operator on `bool` types should already give what we want:
```python3
import numpy as np
a = np.array([1, 1, 0, 0], dtype=bool)
b = np.array([1, 0, 1, 0], dtype=bool)
a + b
```
Gives:
```
array([ True,  True,  True, False])
```